api/v1beta3/cloudstackfailuredomain_types.go

```diff
@@ -72,6 +72,10 @@ type Network struct {
 	// Cloudstack VPC the network belongs to.
 	// +optional
 	VPC *VPC `json:"vpc,omitempty"`
+
+	// Cloudstack Network's routing mode.
+	// +optional
+	NetworkMode string `json:"networkMode,omitempty"`
```

Comment:
To be consistent with cloudstack terminology, I suggest to rename to `RoutingMode`.

network mode: NATTED (VR as Source NAT), ROUTED (no Source NAT)
routing mode (only available for ROUTED network): Dynamic or Static

it would be good to add a line of explanation, for example like
`Routing mode can be Dynamic, or Static. empty value means the network mode is NATTED, not ROUTED.`